app/api/items/ask_hint.go

```diff
@@ -177,38 +176,22 @@ func (requestData *AskHintRequest) UnmarshalJSON(raw []byte) error {
 	if err := requestData.TaskToken.UnmarshalString(*wrapper.TaskToken); err != nil {
 		return fmt.Errorf("invalid task_token: %s", err.Error())
 	}
-	if wrapper.HintRequestedToken == nil {
-		return errors.New("missing hint_requested")
-	}
 	return requestData.unmarshalHintToken(&wrapper)
 }
 
 func (requestData *AskHintRequest) unmarshalHintToken(wrapper *askHintRequestWrapper) error {
-	var platformInfo struct {
-		UsesTokens bool   `gorm:"column:bUsesTokens"`
-		PublicKey  string `gorm:"column:sPublicKey"`
+	if wrapper.HintRequestedToken == nil {
+		return errors.New("missing hint_requested")
 	}
-	var err error
-	if err = requestData.store.Platforms().Select("bUsesTokens, sPublicKey").
-		Joins("JOIN items ON items.idPlatform = platforms.ID").
-		Where("items.ID = ?", requestData.TaskToken.Converted.LocalItemID).
-		Scan(&platformInfo).Error(); gorm.IsRecordNotFoundError(err) {
-		return fmt.Errorf("cannot find the platform for item %s", requestData.TaskToken.LocalItemID)
+
+	err := token.UnmarshalDependingOnItemPlatform(requestData.store, requestData.TaskToken.Converted.LocalItemID,
```

Comment:
UnmarshalDependingOnItemPlatform() is needed just to reuse the code in saveGrade. Score tokens are very similar to hint tokens since they are signed with the task platform's private key too.